### PR TITLE
✨ Source BambooHR - Add Time Off Request Streams

### PR DESCRIPTION
**Pull request by KimPlv**
## What
Add a new streams in Source BambooHR 

## How
- [Time Off Requests](https://documentation.bamboohr.com/reference/time-off-get-time-off-requests-1) full-refresh and incremental 

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- Connector version is set to `0.1.0`
    - `Dockerfile` has version `0.1.0`
- Documentation updated
    - `docs/integrations/source/bamboo-hr.md` including changelog with an entry for the initial version. 
    - `docs/integrations/README.md`

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      "type": ["null", "object"]
+    },
+    "amount": {
+      "type": ["null", "object"]
+    },
+    "actions": {
+      "type": ["null", "object"]
+    },
+    "notes": {
+      "type": ["null", "object"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json:44]:
```suggestion
      "type": ["null", "object"],
      "additionalProperties": true,
      "properties": {}
```

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json:44]: You need to make all objects like this. It is a requirement from the Airbyte cDK

File: airbyte-integrations/connectors/source-bamboo-hr/metadata.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: 90916976-a132-4ce9-8bce-82a03dd58788
-  dockerImageTag: 0.3.5
+  dockerImageTag: 0.3.6
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/metadata.yaml:13]: You're adding a new stream and parameter, please bump the image to `0.4.0`. Don't forget to run `poetry lock` inside the bamboo connector folder to generate the poetry lock file.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    "end": {
+      "type": ["null", "string"],
+      "default": ""
+    },
+    "created": {
+      "type": ["null", "string"],
+      "default": ""
+    },
+    "status": {
+      "type": ["null", "object"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json:28]: You miss this object here. Can you add the proerties and additionalProperties?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json (hunk 3 above)
Comment (KimPlv) [reply airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/schemas/time_off_requests.json]: thanks ! done

File: airbyte-integrations/connectors/source-bamboo-hr/metadata.yaml [hunk 4]
```diff
@@ -10,7 +10,7 @@ data:
   connectorSubtype: api
   connectorType: source
   definitionId: 90916976-a132-4ce9-8bce-82a03dd58788
-  dockerImageTag: 0.3.6
+  dockerImageTag: 0.4.0
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-bamboo-hr/metadata.yaml:13]: Still miss the update in the `pyproject.toml` 


```
[tool.poetry]
version = "0.4.0"
name = "source-bamboo-hr"
```

@KimPlv is it possible to allow maintainers to make changes?

```
 git push                                                                                                                                                                                             ─╯
ERROR: Permission to sendinblue/airbyte.git denied to marcosmarxm.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-bamboo-hr/metadata.yaml (hunk 4 above)
Comment (KimPlv) [reply airbyte-integrations/connectors/source-bamboo-hr/metadata.yaml:13]: Hi @marcosmarxm i give you some access, tell me if it's ok for you

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-bamboo-hr/metadata.yaml:13]: I don't why but still don't have permission. Can you do the changes for me?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Review by marcosmarxm — APPROVED** [review]
LGTM

**Comment by marcosmarxm** [comment]
Hello your feedback matters a lot to us. Can you spare just 2 minutes to complete a [survey](https://form.typeform.com/to/DOBqejte)? We’re dedicated to making the contributor experience even better, and your input is key to achieving excellence. Thank you for helping us improve!

**Pull request merged**